### PR TITLE
[storage] support pagination by adding prefix state value iterator

### PR DESCRIPTION
**Pull request by lightmark**
### Description

I opened a new PR for https://github.com/aptos-labs/aptos-core/pull/5304 as it involves more unexpected changes due to a small issue we found in our state key serialization.

1. To support pagination, I added a prefix scan iterator to facilitate the pagination api.
2. add two helper functions in api side to use later by pagination implementation.

### Test Plan
unit test.

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/5353)

### CODE REVIEW

File: types/src/access_path.rs [hunk 1]
```diff
@@ -63,27 +63,27 @@ impl AccessPath {
...
-        let path = AccessPath::resource_access_vec(key.type_);
+        let path = AccessPath::resource_path_vec(key.type_);
         AccessPath {
             address: key.address,
             path,
         }
     }
 
-    fn code_access_path_vec(key: ModuleId) -> Vec<u8> {
+    pub fn code_path_vec(key: ModuleId) -> Vec<u8> {
```
Comment (msmouse) [inline types/src/access_path.rs:80]: old name seems better..

File: types/src/access_path.rs [hunk 2]
```diff
@@ -63,27 +63,27 @@ impl AccessPath {
         AccessPath { address, path }
     }
 
-    pub fn resource_access_vec(tag: StructTag) -> Vec<u8> {
+    pub fn resource_path_vec(tag: StructTag) -> Vec<u8> {
```
Comment (msmouse) [inline types/src/access_path.rs:66]: `resource_access_path_vec` isn't too verbose to me..

Comment (lightmark) [reply types/src/access_path.rs:66]: Just wanna make `resource` and `module` functions consistent.

File: types/src/access_path.rs (hunk 1 above)
Comment (lightmark) [reply types/src/access_path.rs:80]: ditto

**Review by msmouse — APPROVED** [review]

File: api/src/context.rs [hunk 3]
```diff
@@ -236,10 +239,111 @@ impl Context {
...
+        if iter.next().transpose()?.is_some() {
+            bail!("Too many state items under account ({:?}).", address);
+        }
+        Ok(kvs)
+    }
+
+    pub fn get_resources_by_pagination(
+        &self,
+        address: AccountAddress,
+        prev_state_key: Option<&StateKey>,
```
Comment (CapCap) [inline api/src/context.rs:260]: is this expected from a restful interface input?

Comment (banool) [reply api/src/context.rs:260]: What do you mean? Like the StateKey type? This function itself isn't the endpoint handler. In my PR on top of this one I add StateKeyWrapper that represents this as a hex string in the interface.

**Review by banool — APPROVED** [review]

**Pull request merged**